src/engagement_db_to_analysis/configuration.py

```diff
@@ -1,25 +1,31 @@
 from dataclasses import dataclass
```

Comment:
I don't think we need this since you've documented the classes

Reply:
agreed, I've removed all these annotations now